client/index.js

```diff
@@ -19,31 +19,47 @@ if (!window.Promise) {
 
 const {
   __NEXT_DATA__: {
-    component,
-    errorComponent,
     props,
     err,
     pathname,
-    query
+    query,
+    buildId
   },
   location
 } = window
 
-const Component = evalScript(component).default
-const ErrorComponent = evalScript(errorComponent).default
-let lastAppProps
-
-export const router = createRouter(pathname, query, getURL(), {
-  Component,
-  ErrorComponent,
-  err
-})
+const pageLoader = window.__NEXT_PAGE_LOADER__ = new PageLoader(buildId)
+if (window.__NEXT_LOADED_PAGES__) {
+  window.__NEXT_LOADED_PAGES__.forEach((fn) => fn())
+  delete window.__NEXT_LOADED_PAGES__
```

Comment:
Not sure if this matters anymore, but at some point it was considered "bad" to use `delete` since it is hard to optimize by V8 and other compilers. Whenever I see delete, usually I think there might be a better way via re-assignment. Obviously not a blocker, just an observation